ocaml/xapi-idl/xen/xenops_types.ml

```diff
@@ -77,6 +77,8 @@ module Vm = struct
 
   let default_firmware = Bios [@@deriving rpcty]
 
+  type tpm_type = Vtpm [@@deriving rpcty, sexp]
```

Comment:
A type with type it its name? Is `tpm` enough (or we could use something longer because it is less frequently used: `trusted_platform`). I see tpm and vtpm being used. Is there anything other than vtpm?

Reply:
I'm using `tpm_type` here because `tpm` is already used for the field for the record.

>  Is there anything other than vtpm?
In theory a hardware TPM could be passed through, in this variant is where more information will be attached so the vTPM can be initialized with the correct information when the contents are stored in the database.

We might have to be careful here to avoid getting stale vtpm data into xenopsd, although usually when building this metadata the domain will not be running.